frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseResetAction.java

```diff
@@ -0,0 +1,209 @@
+import com.virtuslab.gitmachete.backend.api.ILocalBranchReference;
+import com.virtuslab.gitmachete.frontend.actions.dialogs.ResetBranchToRemoteInfoDialog;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+@CustomLog
+public abstract class BaseResetAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IBranchNameProvider {
+
+  public static final String SHOW_RESET_INFO = "git-machete.reset.info.show";
+
+  private static final String VCS_NOTIFIER_TITLE = getString(
+      "action.GitMachete.BaseResetToRemoteAction.notification.title");
+
+  @Override
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  @UIEffect
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    log().debug("Performing");
+
+    val project = getProject(anActionEvent);
+    val gitRepository = getSelectedGitRepository(anActionEvent);
+    val branchName = getNameOfBranchUnderAction(anActionEvent);
+    val macheteRepository = getGitMacheteRepositorySnapshot(anActionEvent);
+
+    if (gitRepository == null) {
+      VcsNotifier.getInstance(project).notifyWarning(/* displayId */ null, VCS_NOTIFIER_TITLE,
+          "Skipping the action because no Git repository is selected");
+      return;
+    }
+
+    if (branchName == null) {
+      VcsNotifier.getInstance(project).notifyError(/* displayId */ null, VCS_NOTIFIER_TITLE,
+          "Internal error occurred. For more information see IDE log file");
+      return;
+    }
+
+    if (macheteRepository == null) {
+      VcsNotifier.getInstance(project).notifyError(/* displayId */ null, VCS_NOTIFIER_TITLE,
+          "Internal error occurred. For more information see IDE log file");
+      return;
+    }
+
+    val localBranch = getManagedBranchByName(anActionEvent, branchName);
+    if (localBranch == null) {
+      VcsNotifier.getInstance(project).notifyError(/* displayId */ null, VCS_NOTIFIER_TITLE,
+          "Cannot get local branch '${branchName}'");
+      return;
+    }
+
+    val targetBranch = getTargetBranch(anActionEvent);
+    if (targetBranch == null) {
+      val message = "The target branch for resetting '${localBranch.getName()}' to does not exist, so cannot be reset";
+      log().warn(message);
+      VcsNotifier.getInstance(project).notifyWarning(/* displayId */ null, VCS_NOTIFIER_TITLE, message);
+      return;
+    }
+
+    if (PropertiesComponent.getInstance().getBoolean(SHOW_RESET_INFO, /* defaultValue */ true)) {
+
+      String currentCommitSha = localBranch.getPointedCommit().getHash();
+      if (currentCommitSha.length() == 40) {
+        currentCommitSha = currentCommitSha.substring(0, 15);
+      }
+      val dialogBuilder = MessageDialogBuilder.okCancel(
+          getString("action.GitMachete.BaseResetToRemoteAction.info-dialog.title"),
```

Comment:
Probably `BaseResetAction` to be used everywhere... although TBD if `BaseResetAction` is even needed (once `ResetToParent` is gone)

Reply:
BaseResetAction won't be needed then.